### ISSUE TITLE
macOS Crashing on startup

### ISSUE THREAD

**Issue by Mochaka** [iwalton3/jellyfin-mpv-shim#20]
I'm running macOS Catalina, i installed this app using the instructions you outlined, and it just crashes on startup.

```
jellyfin-mpv-shim                                                              ✔  12:58:56 
❯ jellyfin-mpv-shim
2020-02-18 12:58:58,070 [    INFO] Loaded settings from json: /Users/aaron/Library/Application Support/jellyfin-mpv-shim/conf.json
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
objc[52375]: +[NSCharacterSet initialize] may have been in progress in another thread when fork() was called.
objc[52375]: +[NSCharacterSet initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
2020-02-18 12:58:59,350 [   DEBUG] STREAM b'IHDR' 16 13
2020-02-18 12:58:59,351 [   DEBUG] STREAM b'zTXt' 41 1315
2020-02-18 12:58:59,351 [   DEBUG] STREAM b'bKGD' 1368 6
2020-02-18 12:58:59,352 [   DEBUG] b'bKGD' 1368 6 (unknown)
2020-02-18 12:58:59,352 [   DEBUG] STREAM b'pHYs' 1386 9
2020-02-18 12:58:59,353 [   DEBUG] STREAM b'tIME' 1407 7
2020-02-18 12:58:59,353 [   DEBUG] b'tIME' 1407 7 (unknown)
2020-02-18 12:58:59,354 [   DEBUG] STREAM b'IDAT' 1426 612
objc[52374]: +[NSApplication initialize] may have been in progress in another thread when fork() was called.
objc[52374]: +[NSApplication initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
```

**Comment by iwalton3** [comment]
It sounds like multiprocessing is breaking. https://bugs.python.org/issue33725

Does the application work if you run it with the GUI disabled? (See the instructions in the README about how to configure this.)

You could also try cloning the repository anf adding multiprocessing. set_start_method('forkserver') after multiprocessing.freeze_support() in run.py. Then start the application by running run.py.

@theg00s3 was able to get the GUI working, but I’m not sure if they are on Catalina. The instructions are based on that issue, but as I said in the README they are untested. https://github.com/iwalton3/jellyfin-mpv-shim/issues/18

**Comment by theg00s3** [comment]
I'm running Catalina

**Comment by Mochaka** [comment]
I can confirm that setting the start method does fix it. I'm also running Catalina.

**Comment by iwalton3** [comment]
Ok. I've pushed a patch to master. Could you download the project again and check that the issue is fixed before I release it?

**Comment by Mochaka** [comment]
Can confirm it is now working

Installed using pip3 install git+https://github.com/iwalton3/jellyfin-mpv-shim

Thanks!

**Comment by iwalton3** [comment]
The fix has been released in 1.3.9.
